examples/platform/linux/Options.cpp

```diff
@@ -774,6 +792,35 @@ bool HandleOption(const char * aProgram, OptionSet * aOptions, int aIdentifier,
         LinuxDeviceOptions::GetInstance().tcRequired.SetValue(static_cast<uint16_t>(atoi(aValue)));
         break;
     }
+#endif
+#if CHIP_CONFIG_ENABLE_ICD_SERVER
+    case kDeviceOption_icdActiveModeDurationMs: {
```

Comment:
switch case doesn't validate that the ActiveModeDuration <= to IdleModeDuration; IIRC there is a req in the spec for this.

It is possible for the ICDManager to refuse a configuration even if the inupts are accepted.

Reply:
Suggest adding a check

Reply:
Since the argument parsing is tricky (both are optional and the parsing isn't ordered), I check the arguments when calling `SetModeDuration`

```
+    if (LinuxDeviceOptions::GetInstance().icdActiveModeDurationMs.HasValue() ||
+        LinuxDeviceOptions::GetInstance().icdIdleModeDurationMs.HasValue())
+    {
+        err = Server::GetInstance().GetICDManager().SetModeDurations(LinuxDeviceOptions::GetInstance().icdActiveModeDurationMs,
+                                                                     LinuxDeviceOptions::GetInstance().icdIdleModeDurationMs);
+        if (err != CHIP_NO_ERROR)
+        {
+            ChipLogError(NotSpecified, "Invalid arguments to set ICD mode durations");
+            SuccessOrExit(err);
+        }
+    }
```